src/middleware.ts

```diff
@@ -0,0 +1,27 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+import { languages, defaultLocale } from './utils/getLang';
+
+export function middleware(request: NextRequest) {
+    const acceptLanguage = request.headers.get('Accept-Language');
+
+    if (
+        !acceptLanguage ||
+        request.nextUrl.pathname.startsWith('/api') ||
+        request.nextUrl.pathname.startsWith('/_next') ||
+        request.nextUrl.pathname.startsWith('/favicon')
+    ) {
+        return NextResponse.next();
+    }
+
+    const localeFromHeader = acceptLanguage.slice(0, 2);
+
+    if (languages.some((language) => request.url.includes(`/${language}/`)) || localeFromHeader === defaultLocale) {
```

Comment:
Not `includes`, but `startsWith`

Reply:
No, 'request.url' starts with 'https...'